### ISSUE TITLE
Ladder steps are not sorted (test on binance spot)

### ISSUE THREAD

**Issue by gaugau3000** [nautechsystems/nautilus_trader#894]
# Bug Report

### Expected Behavior

Ladder steps should be sorted.

### Actual Behavior

Ladder steps are not sorted (test on binance spot)

### Steps to Reproduce the Problem

1. subscribe to subscribe_order_book_deltas on binance spot with BTC USDT and depth of 25
2. loop on self.book.bids and assert(self.book.bids.levels[10].price > self.book.bids.levels[20].price)

Will raise an error.

### Specifications

- OS platform: linux
- Python version: from docker image develop
- `nautilus_trader` version: v1.161.0

Have a nice day.

**Comment by cjdsellers** [comment]
I'm looking at this now, I can reproduce the assert, however the state of the book immediately before the failure looks fine. I'll keep digging

```
2022-12-09T22:40:37.001355686Z [INF] TESTER-001.VolatilityMarketMaker-000: bids         price   asks
----------  -------  ---------
            1263.27  [4.4604]
            1263.26  [17.3962]
            1263.25  [10.2293]
            1263.24  [16.8375]
            1263.23  [3.2266]
            1263.22  [4.247]
            1263.21  [2.0529]
            1263.18  [14.817]
            1263.17  [4.8179]
            1263.16  [17.1183]
            1263.15  [11.87]
            1263.14  [17.4444]
            1263.13  [17.9025]
            1263.12  [3.8098]
            1263.11  [9.9031]
            1263.10  [1.5837]
            1263.09  [4.9366]
            1263.08  [1.5836]
            1263.07  [1.9792]
            1263.06  [0.2162]
            1263.05  [3.959]
            1263.04  [0.01]
            1263.02  [0.3613]
            1263.00  [0.5971]
            1262.97  [23.9707]
[18.2464]   1262.96
[3.853]     1262.95
[2.0]       1262.94
[0.3743]    1262.93
[0.1351]    1262.92
[11.5843]   1262.88
[3.9589]    1262.87
[1.5836]    1262.86
[40.8717]   1262.85
[0.0095]    1262.82
[0.8266]    1262.79  # < -- index 10
[4.8342]    1262.78
[1.3777]    1262.77
[15.8384]   1262.76
[15.9306]   1262.75
[5.1732]    1262.74
[8.7591]    1262.73
[3.6399]    1262.72
[1.9836]    1262.71
[10.0996]   1262.69
[1.3065]    1262.65  # <-- index 20
[1.5911]    1262.63
[11.2819]   1262.57
[3.543]     1262.55
[139.5876]  1262.32
```

**Comment by cjdsellers** [comment]
I found the issue, we were using the following to sort the bid side, which isn't an algorithm which will work correctly for this:
```
heapq._siftdown_max(self.levels, 0, len(self.levels) - 1)
```

I replaced which something much less efficient, but correct:
```
if self.is_reversed:
    # TODO(cs): Temporary sorting strategy to fix #894
    self.levels.append(level)
    self.levels = list(reversed(sorted(self.levels)))
```

Ahead of porting the orderbook to rust, where we'll also get better equality and comparison guarantees by using the raw fixed precision integers for prices.

**Comment by cjdsellers** [comment]
Fixed from 04342682453cef72e81cacf7e7343e260d823b47
